ecdsa/src/verifying.rs

```diff
@@ -161,6 +170,21 @@ where
     SignatureSize<C>: ArrayLength<u8>,
 {
     fn verify_prehash(&self, prehash: &[u8], signature: &Signature<C>) -> Result<()> {
+        cfg_if::cfg_if! {
+            if #[cfg(all(target_os = "zkvm", target_vendor = "succinct"))] {
+                let mut sig = signature.clone();
+                let mut recid = 0u8;
+                if let Some(sig_normalized) = sig.normalize_s() {
+                    sig = sig_normalized;
+                    recid ^= 1;
+                }
+                let recid = RecoveryId::from_byte(recid).expect("recovery ID is valid");
```

Comment:
Can you cycle track this logic and see if this is what's incurring the 20K additional cycles vs. `secp256r1`?